pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```
@@ -72,27 +77,120 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
+				apiExportEndpointSlice,
+				apisv1alpha1.APIExportEndpointSliceURLsReady,
+				apisv1alpha1.ErrorGeneratingURLsReason,
+				"",
+			)
 			return err
 		}
 	}
 	conditions.MarkTrue(apiExportEndpointSlice, apisv1alpha1.APIExportValid)
 
-	if err = r.updateEndpoints(ctx, apiExportEndpointSlice, apiExport); err != nil {
+	// Get selector from Partition
+	var selector labels.Selector
+	if apiExportEndpointSlice.Spec.Partition != "" {
+		partition, err := r.getPartition(logicalcluster.From(apiExportEndpointSlice), apiExportEndpointSlice.Spec.Partition)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Don't keep the endpoints if the Partition has been deleted
+				// and is still referenced
+				apiExportEndpointSlice.Status.APIExportEndpoints = nil
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.PartitionValid,
+					apisv1alpha1.PartitionInvalidReferenceReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					err.Error(),
+				)
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.APIExportEndpointSliceURLsReady,
+					apisv1alpha1.ErrorGeneratingURLsReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					"",
+				)
+				// No need to try again
+				return nil
+			} else {
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.PartitionValid,
+					apisv1alpha1.InternalErrorReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					err.Error(),
+				)
+				conditions.MarkUnknown(
+					apiExportEndpointSlice,
+					apisv1alpha1.APIExportEndpointSliceURLsReady,
+					apisv1alpha1.ErrorGeneratingURLsReason,
+					"",
+				)
+				return err
+			}
+		}
+		selector, err = metav1.LabelSelectorAsSelector(partition.Spec.Selector)
+		if err != nil {
+			conditions.MarkFalse(
+				apiExportEndpointSlice,
+				apisv1alpha1.PartitionValid,
+				apisv1alpha1.PartitionInvalidReferenceReason,
+				conditionsv1alpha1.ConditionSeverityError,
+				err.Error(),
```

didn't you want to hide errors?

---

see above